### PR TITLE
[Serverless] remove tagger for serverless

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Remove tagger dependencies from the serverless build

`comp/dogstatsd/replay/writer.go`
Remove reference to tagger by extracting the writer function to a dedicated file (+ !serverless build flag) into a writer_write.go file

`comp/trace/config/setup.go`
Remove reference to tagger by creating containerSetup
With two implementations
- One for severless (no-op) in setup_serverless.go
- One for non-serverless env in setup_container.go
Both file are protected by exclusive build tags

`pkg/aggregator/aggregator.go`
Remove ref to tagger by creating tagger+ tagger_serverless

`pkg/aggregator/context_resolver.go`
Remove tagger ref
Use enrichTag instead of EnrichTag from previous step

`pkg/aggregator/no_aggregation_stream_worker.go`
Same as previous

+ checkCardinality

### CODE REVIEW

**Comment by sgnn7** [comment]
@maxday Just as a heads up, you may be able to slice out the tagger with this in-review PR as well: https://github.com/DataDog/datadog-agent/pull/21314

File: comp/trace/config/setup.go [hunk 1]
```diff
@@ -134,15 +132,10 @@ func prepareConfig(c corecompcfg.Component) (*config.AgentConfig, error) {
...
-	cfg.ContainerTags = containerTagsFunc
-	cfg.ContainerProcRoot = coreConfigObject.GetString("container_proc_root")
+	containerSetup(cfg, coreConfigObject.GetString("container_proc_root"))
 	return cfg, nil
 }
 
-func containerTagsFunc(cid string) ([]string, error) {
-	return tagger.Tag("container_id://"+cid, collectors.HighCardinality)
-}
-
```
Comment (ahmed-mez) [inline comp/trace/config/setup.go:145]: Can we just stub the `containerTagsFunc` function? It'd be enough to isolate the tagger dependency with minimum code changes.

File: comp/dogstatsd/replay/writer_write_serverless.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+//go:build serverless
+
+package replay
+
+func (tc *TrafficCaptureWriter) writeState() (int, error) {
+	// nothing to do here
+	return 0, nil
```
[the remark this answers is not in the document]
Comment (maxday) [reply comp/dogstatsd/replay/writer_write_serverless.go:12]: We're not using captures in serverless

File: comp/trace/config/setup.go (hunk 1 above)
Comment (maxday) [reply comp/trace/config/setup.go]: ✅  thanks!

**Review by ahmed-mez — APPROVED** [review]
The changes in the trace-agent looks good! Let's just make sure the `new-e2e-apm` jobs pass before merging the PR.

File: comp/trace/config/setup_serverless.go [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build serverless
+
+package config
+
+func containerTagsFunc(cid string) ([]string, error) {
```
Comment (ahmed-mez) [inline comp/trace/config/setup_serverless.go:10]: I think the linter might complain about the unused argument
```suggestion
func containerTagsFunc(_ string) ([]string, error) {
```

**Review by wdhif — APPROVED** [review]

**Review by dinooliva — APPROVED** [review]

File: pkg/aggregator/tagger.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+package aggregator
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/tagger"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/tagset"
+)
+
+func enrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string) {
```
Comment (zhuminyi) [inline pkg/aggregator/tagger.go:16]: These functions are using global instance of tagger which should be migrated to a injected component. 
For example:
```
func enrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string, taggerComp tagger.Component) {
	taggerComp.EnrichTags(tb, udsOrigin, clientOrigin, cardinalityName)
}
```
We can get pass in a global tagger instance (by calling GetTaggerInstance()) at this moment and try to remove GetTaggerInstance in the future.

File: comp/dogstatsd/replay/writer_write_serverless.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vickenty) [reply comp/dogstatsd/replay/writer_write_serverless.go:12]: If so, would it make sense to remove captures entirely? (There is another PR that removes a dependency from captures on serverless).

**Comment by pgimalac** [comment]
For the record I computed some stats for the init duration of the serverless agent on this branch:
```
base:   min=268ms, max=290ms, avg=278.3ms, median=278.5ms
branch: min=257ms, max=271ms, avg=264.2ms, median=264.5ms
```
So removing the tagger (and its dependencies) removes around 15ms of init time duration

**Pull request closed without merging**